### DISCUSSION TITLE
Yale YDF40 - Failed to interview ALWAYS

### DISCUSSION THREAD

**Discussion by gregolin** [Koenkk/zigbee2mqtt#12750] [Devices]
Hi everyone.
I have the Yale YDF40, described as [compatible on Zigbee2mqtt](https://www.zigbee2mqtt.io/devices/YDF40.html), but I haven't had success on that.
The system is able to discover it and starts the interview, but always the interview fails. I tried to remove all others zigbee devices and try it only with this lock, and no success. I also tried to remove and return the battery (lock) and also tried with CC2531 and Sonoff Zigbee 3.0 Plus Dongle, any of them fail and it stops at the same point of the process.
Below is my log of the zigbee-herdsman and the zigbee2mqtt, so if someone could help me with any idea I would be very grateful.

<img width="1095" alt="Screen Shot 2022-06-07 at 8 12 13 PM" src="https://user-images.githubusercontent.com/1585520/172498982-7c1c3cc6-82c1-418a-b7eb-d8d0b7bd7b19.png">
<img width="344" alt="Screen Shot 2022-06-07 at 8 12 04 PM" src="https://user-images.githubusercontent.com/1585520/172498988-6ef4e2fb-1551-4593-b42b-1e3edcc9e53e.png">

[zigbee_herdsman.log](https://github.com/Koenkk/zigbee2mqtt/files/8857081/zigbee_herdsman.log)

[zigbee2mqtt.log](https://github.com/Koenkk/zigbee2mqtt/files/8857082/zigbee2mqtt.log)

**Comment by Koenkk** [comment]
- Besides this error, does the device work correctly?
- Can you provide the complete herdsman debug logging until it says interview failed (the attached one is partial)

**Reply by gregolin** [reply]
> * herdsman

Thanks for your message, Koenkk.
About the device, I can open/close it, but I don't receive any message from it (to work with Node Red, for instance).
Regarding the log, I work with HAOS, and following the steps [here](https://www.zigbee2mqtt.io/guide/usage/debug.html#enabling-logging) I can't determine the container ID. When I run the command `docker ps` to discover the container ID on HA terminal I receive the message "docker: command not found".
Is there any way to capture the whole log? Because this one sent is the available in my screen (in the UI).
Thanks.

**Reply by Koenkk** [reply]
I believe you need to ssh as root: https://developers.home-assistant.io/docs/operating-system/debugging/

**Reply by gregolin** [reply]
It's really strange... I'm logged as root as you can see below, but the command docker doesn't have the command log anymore (see on Available Commands... there are only info and registries).
Maybe something has changed on last HA version???
It's crazy, I don't know how can I get the full log to send it to you :-(

<img width="708" alt="image" src="https://user-images.githubusercontent.com/1585520/172688717-f8a50557-7421-4316-a8ee-26fad65b97ec.png">

**Reply by Koenkk** [reply]
What if you execute `ha docker ps -a`?

**Reply by gregolin** [reply]
<img width="701" alt="Screen Shot 2022-06-08 at 3 27 49 PM" src="https://user-images.githubusercontent.com/1585520/172690220-a75f80b3-a301-4458-9346-a11ba57267fc.png">

**Reply by gregolin** [reply]
Please, take a look if this log has what you need.
Thanks
[zigbee_herdsman_2.log](https://github.com/Koenkk/zigbee2mqtt/files/8864326/zigbee_herdsman_2.log)
.

**Reply by gregolin** [reply]
Hi @Koenkk . Is that log file what you are looking for?
Thanks.

**Reply by Koenkk** [reply]
It is the correct log but still partial. Use `docker logs CONTAINER_ID > /tmp/container.log 2>&1` to grab them all

**Reply by gregolin** [reply]
Hi @Koenkk .
I just reset my HA installation, I made a new Debian + HA installation and only added MQTT and Zigbee2MQTT, and after that I just tried to add the Yale YDF40 lock, and the error persists.
I think now I have the full log, could you check it, please?

PS: Another guy in Brazil told me he has the same problem, and the issue started after [this change](https://github.com/Koenkk/zigbee2mqtt/issues/3290). Before that, according to him, everything worked fine.

Thanks.
[log.txt](https://github.com/Koenkk/zigbee2mqtt/files/8883181/log.txt)

**Reply by Koenkk** [reply]
Exactly what I need, pushed a fix, can you check?

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by gregolin** [reply]
I can't see the branch yet, I'll wait more time.
Just another detail... in my first message/screenshot there are manufacturer and model names correctly, but it came from my script and not from the Z2M Trying to add it at this moment doesn't return anything like the image below:

![Captura de tela de 2022-06-11 09-58-42](https://user-images.githubusercontent.com/1585520/173188945-ad44fb4b-1353-4cca-abe1-a5502d5c877f.png)

**Reply by gregolin** [reply]
@Koenkk , it's almost there.
The interview has been completed, but the manufacturer and model weren't recognized, and the Support status is defined as Unsupported.
I, manually, edited the file "/config/zigbee2mqtt/database.db", changing the modelId to iZBModule01, and after that everything is ready. So I think the only issue now is related to recognize the model/manufacturer or the modelId (I don't know which comes first).
Attached you can see my two logs, (1) during the configuration with some issue and (2) after I changed manually the modelId.
Thanks.

[log0611_1.txt](https://github.com/Koenkk/zigbee2mqtt/files/8884443/log0611_1.txt)
[log0611_2.txt](https://github.com/Koenkk/zigbee2mqtt/files/8884444/log0611_2.txt)
![Captura de tela de 2022-06-11 14-38-12](https://user-images.githubusercontent.com/1585520/173198823-a61f539a-5fad-40f5-b373-72b4b9caa9fd.png)

**Reply by Koenkk** [reply]
In your previous attempts you got a modelID, so I think this was just an unlucky attempt and it may work after trying again.

Do I understand correctly that everything works correctly now?

**Reply by gregolin** [reply]
It depends on what you mean as everything working :-)
I removed it again and tried to pair it (twice) to be sure that everything is working fine, and the issue still happen.
First attempt it couldn't finish the interview.
Second one the interview was ok, but the modelID not, I still had to change it manually.
Summarizing, it is working for me after one or two attempts and changing the modelID manually. See my log attached again if you still want to investigate these issues.
Taking advantage on this topic, I have another Yale lock (YMC 420 D) which doesn't have official support by Z2M, but I did it with an [external script](https://github.com/Koenkk/zigbee2mqtt/issues/8946#issuecomment-1117818080), but of course it would be better as official Z2M integration/support. If it interests you I could help you with the logs about this other lock as well.

[log0612_1.txt](https://github.com/Koenkk/zigbee2mqtt/files/8887107/log0612_1.txt)

**Reply by Koenkk** [reply]
In this log the device pairs ok, it is not recognized since it uses a different modelID which I added now:

```js
2022-06-13T00:35:49.268Z zigbee-herdsman:controller:device:log Interview - completed for device '0x000d6f001516d57d'
2022-06-13T00:35:49.269Z zigbee-herdsman:controller:log Succesfully interviewed '0x000d6f001516d57d'
Zigbee2MQTT:info  2022-06-12 21:35:49: Successfully interviewed '0x000d6f001516d57d', device has successfully been paired
Zigbee2MQTT:warn  2022-06-12 21:35:49: Device '0x000d6f001516d57d' with Zigbee model '06ffff2029' and manufacturer name 'ASSA ABLOY iRevo' is NOT supported, please follow https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html
```

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by gregolin** [reply]
Ok, I'll test later. Thanks a lot.
About the other lock model, any chance to work on it as well?
Thanks.

**Reply by Koenkk** [reply]
Looks that the converter of the YMC 420D lock is still in progress. I will let @cefranco76 wether to make a PR or not.

**Reply by gregolin** [reply]
> In this log the device pairs ok, it is not recognized since it uses a different modelID which I added now:
> 
> ```js
> 2022-06-13T00:35:49.268Z zigbee-herdsman:controller:device:log Interview - completed for device '0x000d6f001516d57d'
> 2022-06-13T00:35:49.269Z zigbee-herdsman:controller:log Succesfully interviewed '0x000d6f001516d57d'
> Zigbee2MQTT:info  2022-06-12 21:35:49: Successfully interviewed '0x000d6f001516d57d', device has successfully been paired
> Zigbee2MQTT:warn  2022-06-12 21:35:49: Device '0x000d6f001516d57d' with Zigbee model '06ffff2029' and manufacturer name 'ASSA ABLOY iRevo' is NOT supported, please follow https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html
> ```
> 
> Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

@Koenkk , I tried to pair it (YDF40) twice and both of them worked like a charm. Both recognition and interview.
Do you have any idea when it will be ready on the official product (not development only)?
Thanks a lot.

**Reply by Koenkk** [reply]
Next release is on 1 July

**Reply by gregolin** [reply]
Hi @Koenkk . Sorry to reopen this topic, but it seems there is any issue between the lock and Z2M after some update, I don't know.
I could investigate more about the issue and it seems Z2M isn't receiveing / sending the correct message about this door lock. As you can see in the log, it shows me correctly when I open internally (line 90 for instance), but when I open it externally with fingerprint, I can't see the action, action_source_name or action_user (next-to-last, 94, for instance). The most strange is that everything works fine when I try to open it with password (line 95). 
I've just removed the lines related to other devices in order to have the file easier to read it.
Any idea what is happening? Also you can see, I've removed the device and repaired it in order to eliminate any external problem.
Thanks.
[log.txt](https://github.com/Koenkk/zigbee2mqtt/files/10261116/log.txt)

**Reply by EvertonCa** [reply]
@gregolin did you fixed the issue you were having? I have almost the same lock (YMF40) and since I bought (2 years ago) I have the same problem with failed interview and intermittent messages sent by the lock.

**Reply by gregolin** [reply]
After some attempts it started working fine, and it is fine nowadays.
I think during this period I had only one unsync which I had to remove ir and re-sync... again, some issues during the process, but after some attempts it was perfectly configured. No issues at this moment.
Thanks.
